app.js

```diff
@@ -0,0 +1,52 @@
+// JS2 week1 - Browser environment, DOM manipulation, DOM event listeners
+
+//  create a form to insert a quiz question. It should take a question and 4 options for answers.
+//There should be a way to mark which answer is correct in the form, but you should only be able to select one correct answer.
+
+//  implement one of the two (or both!):
+
+// a. add a button that randomizes the order of the 4 option inputs once they have been filled in.
+
+// b. color the input for the "correct" answer option in green and the "wrong" ones in red. Make sure it's still readable.
+
+//  optional bonus: make sure the question is not longer than 140 characters.
+
+function randomizeOptions() {
+  var optionsContainer = document
+    .getElementById("quizForm")
+    .querySelectorAll('input[type="text"]');
+  var optionsArray = Array.from(optionsContainer);
+
+  // Shuffle the options array
+  optionsArray.sort(() => Math.random() - 0.5);
+
+  // Update input values with shuffled array
+  optionsArray.forEach((option, index) => {
+    option.value = `Shuffled Option ${index + 1}`;
+  });
+}
+
+document
+  .getElementById("quizForm")
+  .addEventListener("submit", function (event) {
+    event.preventDefault();
+
+    // Reset colors
+    document
+      .querySelectorAll('input[type="text"]')
```

Comment:
If you are using the same DOM query in multiple places you should store it in variable, so it is preformed only once. That is called caching - when browser is looking for DOM elements (whenever you use `querySelector` or `querySelectorAll`) it has a performance impact (espoecially with larger websites) so you want to do such operation once and store it's result in a variable for later use.

Reply:
regarding optimizing DOM queries in my code should i put (querySelectorAll('input[type="text"]') in the variable from row number 17 first or from here row no.36?

Reply:
You would do it like that and define it outside of the function. (name is up to you)
```
const textInputs = document.querySelectorAll('input[type="text"]')
```
`querySelectorAll` is a method on a `document` object, so if you will just write `querySelectorAll` then you will get a reference error and the browes will shout at you that it is undefined. 
Every time we run this method on the document object, the browser goes throug all the elements on the page and looks wether they fit the selector passed to the method. In this case `'input[type="text"]'`. This has a performance cost, so if we can we want to do that only once and keep the reference to the list of elements (nodes) for the later use. rows are called lines :)

Reply:
@julczka  how  should I change it from first line 
````
document`
    .getElementById("quizForm")
    .querySelectorAll('input[type="text"]'); 
``should  I make it global variable and send it as argument to my randomise function?

Reply:
const allTextInputs = document.querySelectorAll('input[type="text"]');

`document.getElementById("quizForm).allTextInputs ??`
something like this?